### ISSUE TITLE
Slow loading of IntelliSense in angular project

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#27039]
From https://github.com/Microsoft/vscode/issues/57017

**TypeScript Version:**  3.1.0-dev.20180907

**Search Terms:**

**Repo**
```
git clone https://github.com/HymanZHAN/DatingApp.git
cd DatingApp
cd DatingApp-SPA
npm install
```

Then open a ts file in vs code and trigger intellisense

**Bug:**
It takes around 6 seconds for the project to load. Here is the log: [tsserver.log](https://github.com/Microsoft/TypeScript/files/2372919/tsserver.log)

This is an improvement over 3.0.3

**Playground Link:** 

**Related Issues:**

**Comment by sheetalkamat** [comment]
- Actual completion doesn't take time:
```
Info 999  [15:22:33.284] request:
    {"seq":6,"type":"request","command":"completionInfo","arguments":{"file":"c:/Users/matb/projects/DatingApp/DatingApp-SPA/src/app/app.component.ts","line":8,"offset":37,"includeExternalModuleExports":true,"includeInsertTextCompletions":true}}
Info 1000 [15:22:33.291] getCompletionData: Get current token: 0
Info 1001 [15:22:33.291] getCompletionData: Is inside comment: 0
Info 1002 [15:22:33.291] getCompletionData: Get previous token 1: 0
Info 1003 [15:22:33.292] getCompletionsAtPosition: isCompletionListBlocker: 1
Info 1004 [15:22:33.346] getCompletionData: Semantic work: 54
Info 1005 [15:22:33.401] getCompletionsAtPosition: getCompletionEntriesFromSymbols: 33
Perf 1006 [15:22:33.444] 6::completionInfo: elapsed time (in milliseconds) 160.0651
```
- Seems like 2-3 seconds spent creating file watchers for files in node_modules folder (that are included in program by module resolution) #26135 could help with that.

- There is exception in there. Need to check if we can repro that.
```
Err 988   [15:22:33.278] Exception on executing command {"seq":2,"type":"request","command":"open","arguments":{"file":"c:/Users/matb/projects/DatingApp/DatingApp-SPA/src/app/app.component.ts","fileContent":"import { Component, OnInit } from '@angular/core';\r\nimport { AuthService } from './_services/auth.service';\r\nimport { JwtHelperService } from '@auth0/angular-jwt';\r\nimport { User } from './_models/user';\r\n@Component({\r\n  selector: 'app-root',\r\n  templateUrl: './app.component.html',\r\n  styleUrls: ['./app.component.css']\r\n})\r\nexport class AppComponent implements OnInit {\r\n  jwtHelper = new JwtHelperService();\r\n\r\n  constructor(private authService: AuthService) {}\r\n\r\n  ngOnInit(): void {\r\n    const token = localStorage.getItem('token');\r\n    const user: User = JSON.parse(localStorage.getItem('user'));\r\n\r\n    if (token) {\r\n      this.authService.decodedToken = this.jwtHelper.decodeToken(token);\r\n    }\r\n\r\n    if (user) {\r\n      this.authService.currentUser = user;\r\n      this.authService.changeMemberPhoto(user.photoUrl);\r\n    }\r\n  }\r\n}\r\n","scriptKindName":"TS","projectRootPath":"c:\\Users\\matb\\projects\\DatingApp\\DatingApp-SPA"}}:

    Debug Failure. False expression.

    Error: Debug Failure. False expression.
    at ProjectService.getOrCreateScriptInfoWorker (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:119086:30)
    at ProjectService.getOrCreateScriptInfoOpenedByClientForNormalizedPath (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:119050:29)
    at ProjectService.openClientFileWithNormalizedPath (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:119249:33)
    at IOSession.Session.openClientFile (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:120882:37)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:120074:31)
    at c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:121542:88
    at IOSession.Session.executeWithRequestId (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:121533:28)
    at IOSession.Session.executeCommand (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:121542:33)
    at IOSession.Session.onMessage (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:121562:35)
    at Interface.<anonymous> (c:\Users\matb\projects\DatingApp\DatingApp-SPA\node_modules\typescript\lib\tsserver.js:122816:27)
    at emitOne (events.js:116:13)
    at Interface.emit (events.js:211:7)
    at Interface._onLine (readline.js:282:10)
    at Interface._normalWrite (readline.js:424:12)
    at Socket.ondata (readline.js:141:10)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at Pipe.onread (net.js:594:20)
```

- Noticing there was getErr queued before completions so it would be timing dependent on as to how long before completion request is sent (if already computing errors (which is not the case in the log) we wont be able to see completion till diagnostics for that file are complete)

**Comment by mjbvz** [comment]
Yes completions are pretty quick after the project is loaded. It's just that first `open` that takes a little while

```
Perf 978  [15:26:12.586] 2::open: async elapsed time (in milliseconds) 6082.5485
```
